test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -808,6 +820,378 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
+					client.RawPatch(
+						types.JSONPatchType,
+						[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec/workloadHints", "value": %s }]`, workloadHints)),
+					),
+				)).ToNot(HaveOccurred())
+				By("Applying changes in performance profile and waiting until mcp will start updating")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+				By("Waiting when mcp finishes updates")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+				annotations := map[string]string{
+					"cpu-load-balancing.crio.io": "disable",
+					"cpu-quota.crio.io":          "disable",
+					"irq-load-balancing.crio.io": "disable",
+					"cpu-c-states.crio.io":       "disable",
+					"cpu-freq-governor.crio.io":  "performance",
+				}
+
+				cpuCount := "2"
+				resCpu := resource.MustParse(cpuCount)
+				resMem := resource.MustParse("100Mi")
+				testpod = pods.GetTestPod()
+				testpod.Namespace = testutils.NamespaceTesting
+				testpod.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resCpu,
+						corev1.ResourceMemory: resMem,
+					},
+				}
+				testpod.Spec.NodeSelector = map[string]string{testutils.LabelHostname: workerRTNodes[0].Name}
+				testpod.Annotations = annotations
+				runtimeClass := components.GetComponentName(profile.Name, components.ComponentNamePrefix)
+				testpod.Spec.RuntimeClassName = &runtimeClass
+
+				By("creating test pod")
+				err = testclient.Client.Create(context.TODO(), testpod)
+				Expect(err).ToNot(HaveOccurred())
+				err = pods.WaitForCondition(testpod, corev1.PodReady, corev1.ConditionTrue, 10*time.Minute)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testpod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed), "Test pod does not have QoS class of Guaranteed")
+
+				By("Getting the container cpuset.cpus cgroup")
+				containerID, err := pods.GetContainerIDByName(testpod, "test")
+				Expect(err).ToNot(HaveOccurred())
+
+				containerCgroup := ""
+				Eventually(func() string {
+					cmd := []string{"/bin/bash", "-c", fmt.Sprintf("find /rootfs/sys/fs/cgroup/cpuset/ -name *%s*", containerID)}
+					containerCgroup, err = nodes.ExecCommandOnNode(cmd, &workerRTNodes[0])
+					Expect(err).ToNot(HaveOccurred())
+					return containerCgroup
+				}, (cluster.ComputeTestTimeout(30*time.Second, RunningOnSingleNode)), 5*time.Second).ShouldNot(BeEmpty(),
+					fmt.Sprintf("cannot find cgroup for container %q", containerID))
+
+				By("Verify powersetting of cpus used by the pod")
+				cmd := []string{"/bin/bash", "-c", fmt.Sprintf("cat %s/cpuset.cpus", containerCgroup)}
+				output, err := nodes.ExecCommandOnNode(cmd, &workerRTNodes[0])
+				Expect(err).ToNot(HaveOccurred())
+				cpus, err := cpuset.Parse(output)
+				targetCpus := cpus.ToSlice()
+				By("Verify the rest of the cpus donot haver powersave settings")
```

Comment:
This is incorrect - the code is not checking whether the remaining cpus "donot have powersave settings" - it is checking that they have the default settings, which are a mix (c-states are enabled and governor is performance).